impress-org/givewp-next-gen
# Feature: add anonymous donation block

## Description



This adds the Anonymous Donation block to our core blocks.

## Affects



- This new block adds a checkbox field that integrates with the core donate controller to update the anonymous property on the donation model
- Updated the list of disallowed field names in the field name validator to take in all model properties as well.
- Added migration step for v2 -> v3 form migration

## Visuals



<img width="1108" alt="Screenshot 2023-08-08 at 3 48 19 PM" src="https://github.com/impress-org/givewp-next-gen/assets/10138447/f0705b57-44bd-4314-912e-9f2295f4e9cf">


<img width="821" alt="Screenshot 2023-08-09 at 11 40 15 AM" src="https://github.com/impress-org/givewp-next-gen/assets/10138447/f38d7fb2-c688-4ca0-bf9a-72687af92740">


## Testing Instructions



- Add an anonymous donation block to a v3 form.
- Interact with the settings
- Go to the front-end and check the box for anonymous
- Complete the donation and check the back-end to make sure it persists.
- You should also see a line on the confirmation page.

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205223433088672
